lua/nordic/colors/init.lua

```diff
@@ -3,13 +3,15 @@ local P = require 'nordic.colors.nordic'
 
 local C = {}
 
-local function build_palette()
-    -- Make a new palette.
-    -- Copy the original palette this overrides C.
-    C = vim.deepcopy(P)
-    -- Because of the override we need to re-add the build_palette to the C table.
-    C.build_palette = build_palette
-    -- If we had attached build_palette to C we would lost it after the override.
+function C.build_palette()
+    -- clear all values except build_palette
+    for k, _ in pairs(C) do
+        if k ~= "build_palette" then
+            C[k] = nil
+        end
+    end
+    -- copy all values from the base palette
+    U.merge_inplace(C, vim.deepcopy(P))
```

Comment:
Are you sure we have to deepcopy?

Reply:
Nope I will look into that

Reply:
Not if I change `merge_inplace` to a deep merge:

[feat(utils)!: merge_inplace now deep merges;](https://github.com/AlexvZyl/nordic.nvim/pull/137/commits/dd23021ea48defb129a4e9360d7669b04dbd2a67)
[refactor(colors): clean up build_palette;](https://github.com/AlexvZyl/nordic.nvim/pull/137/commits/4c697537d020fa2df54c89e3d04f9ad8cc542579)

Reply:
Nice.